source/Octopus.Cli/Commands/Releases/PromoteReleaseCommand.cs

```diff
@@ -44,14 +46,20 @@ public async Task Request()
 
             environment = await Repository.Environments.FindByNameOrIdOrFail(FromEnvironmentNameOrId).ConfigureAwait(false);
 
-            var dashboard = await Repository.Dashboards.GetDynamicDashboard(new[] {project.Id}, new[] {environment.Id}).ConfigureAwait(false);
-            var dashboardItem = dashboard.Items.Where(e => e.EnvironmentId == environment.Id && e.ProjectId == project.Id)
-                .OrderByDescending(i => SemanticVersion.Parse(i.ReleaseVersion))
-                .FirstOrDefault();
+            var dashboard = await Repository.Dashboards.GetDynamicDashboard(new[] {project.Id}, new[] {environment.Id}, true).ConfigureAwait(false);
+            var dashboardItems = dashboard.Items
+                .Where(e => e.EnvironmentId == environment.Id && e.ProjectId == project.Id)
+                .OrderByDescending(i => SemanticVersion.Parse(i.ReleaseVersion));
+
+            var dashboardItem = UseLatestSuccessfulRelease
+                ? dashboardItems.FirstOrDefault(x => x.State == TaskState.Success)
+                : dashboardItems.FirstOrDefault();
 
             if (dashboardItem == null)
             {
-                throw new CouldNotFindException("latest deployment of the project for this environment. Please check that a deployment for this project/environment exists on the dashboard.");
+                var deploymentType = UseLatestSuccessfulRelease ? "successful" : "";
+
+                throw new CouldNotFindException($"latest {deploymentType} deployment of the project for this environment. Please check that a {deploymentType} deployment for this project/environment exists on the dashboard.");
```

Comment:
This message will have too many spaces when `UseLatestSuccessfulRelease == false`